mutable_tree.go

```diff
@@ -137,7 +137,9 @@ func (tree *MutableTree) Hash() []byte {
 
 // WorkingHash returns the hash of the current working tree.
 func (tree *MutableTree) WorkingHash() []byte {
-	return tree.root.hashWithCount(tree.WorkingVersion())
+	// Use tree.version instead of tree.WorkingVersion() because tree.version
+	// is always the latest version
+	return tree.root.hashWithCount(tree.version + 1)
```

Comment:
just saying out loud here that this must've been included in the live data we have or we would have gotten app hash mismatches with mainnet?

Reply:
Cosmos v0.47.x doesn't use the WorkingHash/WorkingVersion afaik, they are just defined on the iavl interface and unused by the rootmultistore

Reply:
right - this mostly updated to ensure the tree tests pass and keep the behavior consistent.

I thought it could be a potential issue with rollbacks via `LoadVersionForOverwriting` - two cases:
1. Not possible via "regular" rollbacks since you can't load `InitialVersion-1` to write `InitialVersion` again
2. Creating a new tree with an existing node database, which is dependent on cosmos sdk behavior. If the node database still exists for this tree and we create a new tree at InitialVersion, this would hit the codepath in `SaveVersion()` that calls `WorkingHash()` that produces an incorrect mismatch

Added a test for the second case to make sure we aren't depending on cosmos handling it properly and WorkingHash() is as expected in that case

Reply:
Right, I forgot about the `WorkingHash` within those methods 🤦.  I read through the `LoadVersionForOverwriting` test, nice comments there too -- some great thinking into how the iavl library is used 💯